apps/ledger-live-desktop/src/renderer/components/WebPTXPlayer/CustomHandlers.ts

```diff
@@ -57,10 +59,10 @@ export function usePTXCustomHandlers(manifest: WebviewProps["manifest"]) {
                 ...exchangeParams,
                 exchangeType: ExchangeType[exchangeParams.exchangeType],
                 onResult: (nonce: string) => {
-                  onSuccess(nonce);
+                  onSuccess(nonce, device);
```

Comment:
Not sure how much it keeps the device in sync here because it will be hoisted when the `custom.exchange.start` is called it could be the old value

It would probably be better if `onResult` and `onCancel` could return the device as it's done for LLM
I checked the code and it seems pretty straightforward to add some data:
- Inputs: https://github.com/LedgerHQ/ledger-live/blob/develop/libs/ledger-live-common/src/exchange/platform/startExchange.ts#L16
- Success: https://github.com/LedgerHQ/ledger-live/blob/develop/libs/ledger-live-common/src/exchange/platform/startExchange.ts#L49-L52
- Error: https://github.com/LedgerHQ/ledger-live/blob/develop/libs/ledger-live-common/src/exchange/platform/startExchange.ts#L62-L65
- StartExchange inputs: https://github.com/LedgerHQ/ledger-live/blob/develop/libs/ledger-live-common/src/hw/actions/startExchange.ts#L150-L155
- startExchange mapResult: https://github.com/LedgerHQ/ledger-live/blob/develop/libs/ledger-live-common/src/hw/actions/startExchange.ts#L37-L49

Another solution could be to simply use `const device = useSelector(getCurrentDevice);` in `apps/ledger-live-desktop/src/renderer/components/LiveAppDrawer.tsx` because if we do the changes above we might want to update LLM so result.device comes from the action and not a state.

You can also cleanup [this types](https://github.com/LedgerHQ/ledger-live/blob/develop/apps/ledger-live-desktop/src/renderer/components/LiveAppDrawer.tsx#L153-L161) that are not needed anymore
<img width="1456" alt="Screenshot 2024-05-06 at 16 08 00" src="https://github.com/LedgerHQ/ledger-live/assets/3009753/438588d4-8845-44dd-9698-172796303d19">

Reply:
Hey not sure we have a lot of documentation on the device actions and exchange part to be honest
I'm pretty familiar with this because I worked on some issues around it

My main concern here is to avoid having to much differences between LLD and LLM and sharing as much as possible, especially because you have a lot of ways to pass the data in the end correctly